.github/workflows/bump.yml

```diff
@@ -20,15 +26,97 @@ jobs:
-      
       - name: merge and write Changelogs
         id: merge-changelogs
         uses: sap/cloud-sdk-js/.github/actions/merge-and-write-changelogs@main
         env:
           VERSION: ${{ steps.bump.outputs.version }}
-
       - uses: sap/cloud-sdk-js/.github/actions/commit-and-tag@main
         name: Commit and tag
         with:
           version: ${{ steps.bump.outputs.version }}
           user-name: sap-ai-sdk
+
+  prepare-release-notes:
+    name: Prepare Release Notes
+    runs-on: ubuntu-latest
+    needs: [bump]
+    outputs:
+      pr_url: ${{ steps.open_pr.outputs.pr_url }}
+    steps:
+      - name: Checkout current repository
+        uses: sap/ai-sdk-js/.github/actions/setup@main
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Checkout Docs Repo
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.DOCS_REPO }}
+          token: ${{ secrets.BOT_SDK_JS_FOR_DOCS_REPO_PR }}
+          path: ${{ env.DOCS_CHECKOUT_PATH }}
+          fetch-depth: 0
+      - name: Update release notes file
+        id: update-notes
+        run: |
+          NODE_OPTIONS="--loader ts-node/esm" pnpm ts-node --esm scripts/execute-add-changelog.ts
+      - name: Open Release notes PR
+        id: open_pr
+        working-directory: ${{ env.DOCS_CHECKOUT_PATH }}
+        run: |
+          git config --local user.email "cloudsdk@sap.com"
+          git config --local user.name "SAP Cloud SDK Bot"
+          BRANCH_NAME="update-release-notes-${{ needs.bump.outputs.version }}"
+          git checkout -b "${BRANCH_NAME}"
+          git add docs-js/release-notes.mdx
+          if git diff --staged --quiet; then
+            echo "No changes to commit."
+            # Output an empty pr_url if no changes
+            echo "pr_url=" >> "$GITHUB_OUTPUT"
+          else
+            git commit -m "update release notes" -a
+            git push -u origin "${BRANCH_NAME}"
+
+            echo ${{ secrets.BOT_SDK_JS_FOR_DOCS_REPO_PR }} | gh auth login --hostname github.com --with-token
+            gh config set prompt disabled
+            PR_BODY="Auto-created by update release notes workflow."
+            PR_TITLE="Update JS Release Notes"
+            PR_URL=$(gh pr create --head "${BRANCH_NAME}" --title "${PR_TITLE}" --body "${PR_BODY}")
+            echo "Release notes PR created: $PR_URL"
+            if [[ -n "$PR_URL" ]]; then
+              echo "pr_url=$PR_URL" >> "$GITHUB_OUTPUT"
+            else
+              echo "Failed to create PR or extract URL."
+              echo "pr_url=" >> "$GITHUB_OUTPUT"
+            fi
+          fi
+
+  handle-failure:
+    name: Handle Failure
+    runs-on: ubuntu-latest
+    needs: [bump, prepare-release-notes]
+    if: ${{ failure() }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: 'Delete Release Tag'
```

Comment:
[q] Why are we removing the tag? New versions should have been already committed to the main branch. Delete the tag won't really revert the changes and also feels unnecessary.

Reply:
The idea was to revert the commit also initially along with deleting the tag, but as we discussed yesterday, commit and tag is the last step of the job, so it doesn't make sense to do this now, as commit and tag is only done if everything ran successfully. So removed the `handle-failure` step completely.